src/test/java/com/iexec/worker/compute/ComputeControllerTests.java

```diff
@@ -71,35 +74,77 @@ void sendExitCauseForComputeComputeStage() {
-        Assertions.assertTrue(response.getStatusCode().is2xxSuccessful());
-        Assertions.assertEquals(HttpStatus.OK.value(), response.getStatusCode().value());
+        Assertions.assertTrue(singleResponse.getStatusCode().is2xxSuccessful());
+        Assertions.assertEquals(HttpStatus.OK.value(), singleResponse.getStatusCode().value());
+
+        // Test single cause via new bulk endpoint
+        final ResponseEntity<Void> bulkSingleResponse = computeController.sendExitCausesForGivenComputeStage(
+                AUTH_HEADER,
+                ComputeStage.POST,
+                CHAIN_TASK_ID,
+                Collections.singletonList(CAUSE)
+        );
+        Assertions.assertEquals(HttpStatus.OK.value(), bulkSingleResponse.getStatusCode().value());
+
+        // Test multiple causes via new bulk endpoint
+        List<ReplicateStatusCause> multipleCauses = Arrays.asList(
+                ReplicateStatusCause.PRE_COMPUTE_DATASET_URL_MISSING,
+                ReplicateStatusCause.PRE_COMPUTE_INVALID_DATASET_CHECKSUM
+        );
+        final ResponseEntity<Void> bulkMultipleResponse = computeController.sendExitCausesForGivenComputeStage(
+                AUTH_HEADER,
+                ComputeStage.PRE,
+                CHAIN_TASK_ID,
+                multipleCauses
+        );
+        Assertions.assertEquals(HttpStatus.OK.value(), bulkMultipleResponse.getStatusCode().value());
     }
 
     @Test
-    void shouldNotSendExitCauseForComputeComputeStageSinceNoCause() {
+    void shouldReturnBadRequestForInvalidExitCauses() {
         when(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(CHAIN_TASK_ID, AUTH_HEADER))
                 .thenReturn(true);
 
-        final ResponseEntity<?> response =
+        // Test null cause via deprecated endpoint
+        final ResponseEntity<?> nullCauseResponse =
                 computeController.sendExitCauseForGivenComputeStage(AUTH_HEADER,
                         ComputeStage.PRE,
                         CHAIN_TASK_ID,
                         new ExitMessage(null));
-        Assertions.assertEquals(HttpStatus.BAD_REQUEST.value(), response.getStatusCode().value());
+        Assertions.assertEquals(HttpStatus.BAD_REQUEST.value(), nullCauseResponse.getStatusCode().value());
+
+        // Test null causes list via new bulk endpoint
+        final ResponseEntity<Void> nullListResponse = computeController.sendExitCausesForGivenComputeStage(
+                AUTH_HEADER,
+                ComputeStage.PRE,
+                CHAIN_TASK_ID,
+                null
+        );
+        Assertions.assertEquals(HttpStatus.BAD_REQUEST.value(), nullListResponse.getStatusCode().value());
+
+        // Test empty causes list via new bulk endpoint
+        final ResponseEntity<Void> emptyListResponse = computeController.sendExitCausesForGivenComputeStage(
+                AUTH_HEADER,
+                ComputeStage.PRE,
+                CHAIN_TASK_ID,
+                Collections.emptyList()
```

Comment:
For more concise imports, you can use
```suggestion
                List.of()
```

Reply:
Resolved in [a80e663](https://github.com/iExecBlockchainComputing/iexec-worker/pull/653/commits/a80e6637510785b3b500570dc702721b7a4f1d35)